x-pack/test/apm_api_integration/tests/service_groups/service_group_count.spec.ts

```diff
@@ -0,0 +1,206 @@
+      .send(bulkDeleteBody);
+  }
+
+  registry.when('Service group counts', { config: 'basic', archives: [] }, () => {
+    let synthbeansServiceGroupId: string;
+    let opbeansServiceGroupId: string;
+    before(async () => {
+      const synthServices = [
+        apm
+          .service({ name: 'synth-go', environment: 'testing', agentName: 'go' })
+          .instance('instance-1'),
+        apm
+          .service({ name: 'synth-java', environment: 'testing', agentName: 'java' })
+          .instance('instance-2'),
+        apm
+          .service({ name: 'opbeans-node', environment: 'testing', agentName: 'nodejs' })
+          .instance('instance-3'),
+      ];
+
+      const [, { body: synthbeansServiceGroup }, { body: opbeansServiceGroup }] = await Promise.all(
+        [
+          synthtraceEsClient.index(
+            synthServices.map((service) =>
+              timerange(start, end)
+                .interval('5m')
+                .rate(1)
+                .generator((timestamp) =>
+                  service
+                    .transaction({
+                      transactionName: 'GET /api/product/list',
+                      transactionType: 'request',
+                    })
+                    .duration(2000)
+                    .timestamp(timestamp)
+                    .children(
+                      service
+                        .span({
+                          spanName: '/_search',
+                          spanType: 'db',
+                          spanSubtype: 'elasticsearch',
+                        })
+                        .destination('elasticsearch')
+                        .duration(100)
+                        .success()
+                        .timestamp(timestamp),
+                      service
+                        .span({
+                          spanName: '/_search',
+                          spanType: 'db',
+                          spanSubtype: 'elasticsearch',
+                        })
+                        .destination('elasticsearch')
+                        .duration(300)
+                        .success()
+                        .timestamp(timestamp)
+                    )
+                    .errors(
+                      service.error({ message: 'error 1', type: 'foo' }).timestamp(timestamp),
+                      service.error({ message: 'error 2', type: 'foo' }).timestamp(timestamp),
+                      service.error({ message: 'error 3', type: 'bar' }).timestamp(timestamp)
+                    )
+                )
+            )
+          ),
+          saveServiceGroup({
+            groupName: 'synthbeans',
+            kuery: 'service.name: synth*',
+          }),
+          saveServiceGroup({
+            groupName: 'opbeans',
+            kuery: 'service.name: opbeans*',
+          }),
+        ]
+      );
+      synthbeansServiceGroupId = synthbeansServiceGroup.id;
+      opbeansServiceGroupId = opbeansServiceGroup.id;
+    });
```

Comment:
We normally extract this outside the test to reduce noise. Can we do the same here? Perhaps create a folder `service_group_count` with both the test and the data generation function